k9mail/src/main/java/com/fsck/k9/mailstore/migrations/MigrationTo58.java

```diff
@@ -0,0 +1,31 @@
+package com.fsck.k9.mailstore.migrations;
+
+
+import android.database.sqlite.SQLiteDatabase;
+
+
+class MigrationTo58 {
+    static void cleanUpOrphanedData(SQLiteDatabase db) {
+        cleanUpFtsTable(db);
+        cleanUpMessagePartsTable(db);
+    }
+    
+    private static void cleanUpFtsTable(SQLiteDatabase db) {
+        MigrationTo56.cleanUpFtsTable(db);
+    }
+
+    private static void cleanUpMessagePartsTable(SQLiteDatabase db) {
+        db.execSQL("DELETE FROM message_parts WHERE root NOT IN " +
+                "(SELECT message_part_id FROM messages WHERE deleted = 0)");
```

Comment:
We should try and work out how they are being created with a null ID first, fix that and delete those in a new migration step.